CONTRIBUTING.md

```diff
@@ -18,12 +18,68 @@
 
 8. If your backport PR(s) get labeled with `bp-conflict`, it means they cannot be automatically be merged. You can help get them merged by openening a PR against the already-existing backport branch (will be named something like `mergify/bp/3.5.x/pr-2512`) with the necessary cleanup changes. The admins will merge your cleanup PR and remove the `bp-conflict` label if appropriate.
 
+---
+
+## Building and Testing
+
+### Dependencies
+
+Chisel uses the [Mill](https://mill-build.org/) build tool.
+You can install it as described on the Chisel [installation instructions](https://www.chisel-lang.org/docs/installation), or just use the bootstrap script in this repository: `./mill`.
+Developers should real the Mill documentation to understand the basic commands and use.
+
+The only real dependency for development is the JDK.
+Any JDK 8 or newer will work for most development, but note that developing the CIRCT Panama bindings requires Java 21 or newer.
```

Comment:
```suggestion
Any JDK 8 or newer will work for most development, but note that developing the CIRCT Panama bindings requires Java 21. 
```
Panama API changed in JDK22

Reply:
Good to know. That might be somewhat problematic when we want to support Java 25 (next LTS) next year, but we can punt on that for now.

Reply:
Nice, we will provide have a new branch for migration, and maybe provide due version in the future, since JDK22 breaks the panama API.